addons/web/models/models.py

```diff
@@ -831,6 +835,197 @@ def group_id_name(value):
 
             return { 'values': field_range, }
 
+    def onchange2(self, values: Dict, field_name: str, fields_spec: Dict):
+        """ Perform an onchange on the given field.
+
+            :param values: dictionary mapping field names to values on the form view,
+                giving the current state of modification
+            :param field_name: name of the modified field, or list of field
+                names (in view order), or False
+            :param fields_spec: dictionary specifying the fields in the view,
+                just like the one used by :meth:`web_read`
+
+            When ``field_name`` is falsy, the method first adds default values
+            to ``values``, computes the remaining fields, applies onchange
+            methods to them, and return all the fields in ``fields_spec``.
+        """
+        # this is for tests using `Form`
+        self.env.flush_all()
+
+        env = self.env
+        if isinstance(field_name, list):
+            field_names = field_name
+        elif field_name:
+            field_names = [field_name]
+        else:
+            field_names = []
```

Comment:
could we always make a list of field_names:
- an empty list to have the "default_get" feature
- the list with one field for "real" onchange

It would remove some of the geometry of the function that is not needed

Reply:
Maybe we could have just one field name?
* the values `None` or `False` stand for "no field", and is reserved for the first call (getting defaults)
* a string value for the real onchange
* any other value would be an error

We don't have sensible use-cases for onchange on multiple fields, anyway.

It _could_ make sense _if_ some mechanism in the form view simultaneously assigns _several_ fields. @aab-odoo what do you think?

Reply:
Technically, this mechanism exists. A field/widget can update several fields with a single call. The `update` function of our js model takes a dict in argument (mapping fieldName -> value). So if one of the fieldNames in the dict as an onchange, we do the call. I don't know if we have usecases, but they would be legitimate imo.